modules/passkey/contracts/SafeWebAuthnSignerProxyFactory.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity >=0.8.0;
+
+import {ICustomECDSASignerProxyFactory} from "./interfaces/ICustomECDSASignerProxyFactory.sol";
+import {IP256Verifier} from "./interfaces/IP256Verifier.sol";
+import {ERC1271} from "./libraries/ERC1271.sol";
+import {WebAuthn} from "./libraries/WebAuthn.sol";
+import {SafeWebAuthnSignerProxy} from "./SafeWebAuthnSignerProxy.sol";
+/**
+ * @title SafeWebAuthnSignerProxyFactory
+ * @dev A factory contract for creating and managing WebAuthn proxy signers.
+ */
+contract SafeWebAuthnSignerProxyFactory is ICustomECDSASignerProxyFactory {
+    /**
+     * @inheritdoc ICustomECDSASignerProxyFactory
+     */
+    function getSigner(address singleton, uint256 x, uint256 y, address verifier) public view override returns (address signer) {
+        bytes32 codeHash = keccak256(
+            abi.encodePacked(type(SafeWebAuthnSignerProxy).creationCode, uint256(uint160(singleton)), x, y, uint256(uint160(verifier)))
+        );
+        signer = address(uint160(uint256(keccak256(abi.encodePacked(hex"ff", address(this), bytes32(0), codeHash)))));
+    }
+
+    /**
+     * @inheritdoc ICustomECDSASignerProxyFactory
+     */
+    function createSigner(address singleton, uint256 x, uint256 y, address verifier) external returns (address signer) {
+        signer = getSigner(singleton, x, y, verifier);
+
+        if (_hasNoCode(signer)) {
+            SafeWebAuthnSignerProxy created = new SafeWebAuthnSignerProxy{salt: bytes32(0)}(singleton, x, y, verifier);
+            require(address(created) == signer);
+        }
+    }
+
+    /**
+     * @inheritdoc ICustomECDSASignerProxyFactory
+     */
+    function isValidSignatureForSigner(
+        bytes32 message,
+        bytes calldata signature,
+        uint256 x,
+        uint256 y,
+        address verifier
+    ) external view override returns (bytes4 magicValue) {
+        if (WebAuthn.verifySignature(message, signature, WebAuthn.USER_VERIFICATION, x, y, IP256Verifier(verifier))) {
```

Comment:
This should use the singleton for verification. It ensures that the implementation is the same.

Reply:
done